### PR TITLE
Audit logs script security-529

### PR DESCRIPTION
**Pull request by avinashneelakantam**

### CODE REVIEW

File: logging/src/scripts/audit_logs_script.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  then
+    echo "Downloading ${i} ........."
+    /usr/bin/aws --region us-west-1 rds download-db-log-file-portion --db-instance-identifier ${INSTANCE} --log-file-name ${i} --starting-token 0 --output text > ${FILE}
+
+   fi
+  done
+
+ if [ $? -eq 0 ]
+then
+  echo "Script ran Successfully"
```
Comment (mlipardo) [inline logging/src/scripts/audit_logs_script.sh]: 👍

File: logging/src/scripts/audit_logs_script.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  exit 0
+else
+  echo "Script Failed" >&2
```
Comment (mlipardo) [inline logging/src/scripts/audit_logs_script.sh]: good error handling

Comment (david-sanabria) [reply logging/src/scripts/audit_logs_script.sh]: works for me. Another convention used in scripts is a logical or (`||`) used as an assertion .

Example:
`  somecommand || echo "Step Failed [$?]" >&2`

Useful when you don't need to fail, and only evaluated when the first command returns a non-zero result.

File: logging/src/scripts/Cloud_watch_logs_conf_file_Security_529.conf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[/var/log/postgresql/audit.log]
```
Comment (david-sanabria) [inline logging/src/scripts/Cloud_watch_logs_conf_file_Security_529.conf]: See my comments on this file in separate PR; there are several.

File: logging/src/scripts/audit_logs_script.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+LAST_FILE_RUNDECK=`ls -lrt | grep postgresql | tail -1 | awk '{print $9}'`
+LAST_RUNDECK_FILE=`basename ${i}`
+
+/usr/bin/aws --region us-west-1 rds download-db-log-file-portion --db-instance-identifier ${INSTANCE} --log-file-name ${LAST_FILE_RUNDECK} --starting-token 0 --output text > ${LAST_RUNDECK_FILE}
+
+Last_file_rundeck_re_download_code=`echo "$?"`
+
+if [ ${Last_file_rundeck_re_download_code} -ne 0 ]
+then
+  echo "Last rundeck file download Failed" >&2
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Please add the name of the log file we tried to download, which failed.

File: logging/src/scripts/audit_logs_script.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  exit 2
+else 
+  echo "Last rundeck file download is succesfull"
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Add file name of log file.

File: logging/src/scripts/audit_logs_script.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  if [ ! -f ${FILE} ];
+  then
+    echo "Downloading ${i} ........."
+    /usr/bin/aws --region us-west-1 rds download-db-log-file-portion --db-instance-identifier ${INSTANCE} --log-file-name ${i} --starting-token 0 --output text > ${FILE}
+    
+    Latest_rds_log_download_code=`echo "$?"`
+    
+    if [ ${Latest_rds_log_download_code} -ne 0 ]
+    then
+      echo "RDS file download Failed" >&2
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Add file name.

File: logging/src/scripts/audit_logs_script.sh [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# !/bin/bash
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Global change. Please add a timestamp to the start of all output lines. We need to be able to parse the output of this script as well, and need timestamps.

File: logging/src/scripts/audit_logs_script.sh [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+export INSTANCE=$1
+
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: `TS_FORMAT="YYYY-MM-DDTHH24:MI:SS.SSSSSS"`
logical format given. Use the appropriate "`%`" format codes to get this format.

File: logging/src/scripts/audit_logs_script.sh [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+cd /var/log/postgresql-rds/${INSTANCE}/error
+LAST_FILE_RUNDECK=`ls -lrt | grep postgresql | tail -1 | awk '{print $9}'`
+LAST_RUNDECK_FILE=`basename ${i}`
+
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Please add something like this: `echo "$(date $TS_FORMAT)\tPostgres-RDS Logs Backup Starting\t$INSTANCE\t$0`

And add another entry at the end for "Finished"

File: logging/src/scripts/audit_logs_script.sh [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+Last_file_rundeck_re_download_code=`echo "$?"`
+
+if [ ${Last_file_rundeck_re_download_code} -ne 0 ]
+then
+  echo "Last rundeck file download Failed" >&2
+  exit 2
+else 
+  echo "Last rundeck file download is succesfull"
+fi
+
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Add file counter variable

`FILES_DOWNLOADED=0`

File: logging/src/scripts/audit_logs_script.sh [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    /usr/bin/aws --region us-west-1 rds download-db-log-file-portion --db-instance-identifier ${INSTANCE} --log-file-name ${i} --starting-token 0 --output text > ${FILE}
+    
+    Latest_rds_log_download_code=`echo "$?"`
+    
+    if [ ${Latest_rds_log_download_code} -ne 0 ]
+    then
+      echo "RDS file download Failed" >&2
+      exit 2
+    else 
+    echo "RDS file download is succesfull"
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Add `FILES_DOWNLOADED = $FILES_DOWNLOADED + 1`

File: logging/src/scripts/audit_logs_script.sh [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    fi
+   fi
+  done
+
+
+find /var/log/postgresql-rds/${INSTANCE}/error -type f -mtime +7 -name 'postgresql*' -exec rm {} \;
```
Comment (david-sanabria) [inline logging/src/scripts/audit_logs_script.sh]: Having this command at the end of the file could result in a non-zero return code. Let's end the script with an explicit:
`echo "$(date $TS_FORMAT)\tPostgres-RDS Logs Backup Finished\t$INSTANCE\tLog Files=$FILES_DOWNLOADED`
`exit 0`

File: logging/src/scripts/audit_logs_script.sh (hunk 8 above)
Comment (mlipardo) [reply logging/src/scripts/audit_logs_script.sh]: we use this instead TS_FORMAT="+%Y-%m-%d-%H:%M:%S:%s"

File: logging/src/scripts/audit_logs_script.sh (hunk 12 above)
Comment (mlipardo) [reply logging/src/scripts/audit_logs_script.sh]: added

**Pull request merged**